### PR TITLE
Support XRP X-address

### LINKED ISSUE #729: XRP X-address support
- https://xrpaddress.info/
- https://github.com/ripple/ripple-address-codec

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #729 and a few minor fixes

### CODE REVIEW

File: swift/Sources/Addresses/CoinType+Address.swift [hunk 1]
```diff
@@ -41,7 +40,10 @@ public extension CoinType {
         case .ontology:
             return OntologyAddress(string: string)
         case .xrp:
-            return RippleAddress(string: string)
+            if let address = RippleAddress(string: string) {
+                return address
```
Comment (vikmeup) [inline swift/Sources/Addresses/CoinType+Address.swift]: is there a way to combine into one class? Just `RippleAddress`

File: TrustWalletCore.podspec [hunk 2]
```diff
@@ -132,13 +132,15 @@ Pod::Spec.new do |s|
     "#{protobuf_dir}/src/google/protobuf/port_def.inc",
     "#{protobuf_dir}/src/google/protobuf/port_undef.inc",
   s.exclude_files =
-    'trezor-crypto/src/rand.c'
+    'trezor-crypto/src/rand.c',
+    'trezor-crypto/include/TrezorCrypto/**/base58.h'
   s.public_header_files =
```
Comment (vikmeup) [inline TrustWalletCore.podspec]: any reason for `base58` exclude?

File: swift/Sources/Addresses/CoinType+Address.swift (hunk 1 above)
Comment (hewigovens) [reply swift/Sources/Addresses/CoinType+Address.swift]: App won't notice this change, this is same pattern as we did in BitcoinAddress and SegwitAddress, quite different

Comment (vikmeup) [reply swift/Sources/Addresses/CoinType+Address.swift]: I was just curious in terms of simplicity to handle both in one class, as it's just one address

File: TrustWalletCore.podspec (hunk 2 above)
Comment (hewigovens) [reply TrustWalletCore.podspec]: This tries to fix the warnings when you build wallet core from source (pod -> master), we don't use c base58.h in the app

File: swift/Sources/Addresses/CoinType+Address.swift (hunk 1 above)
Comment (hewigovens) [reply swift/Sources/Addresses/CoinType+Address.swift]: that would work too, have to introduce an new address type

**Pull request merged**